### PR TITLE
Include docs for msghandlers list

### PR DESCRIPTION
**Pull request by rennerocha**
I found that the content of https://docs.bottery.io/en/latest/patterns.html is not updated with the structure that are created after using the _startproject_ command. There is no more a _bot.py_ file, and the messages are handled by a _msghandlers_ list (is this slightly inspired by Django URL confs?)

I needed to dig into the code to understand how this works (and I hope that I understood it correctly :smile: ), so I tried to improve the documentation with the thing that I found.

I don't know if the old method with decorators will still continue being used, so I didn't remove the patterns.rst file . However, it may be removed as well if _msghandlers_ will be the only way to define the rules from now.

Please let me know if something is incorrect (or that will be changed in the nearest future), so I can try to fix it.

### CODE REVIEW

File: docs/handlers.rst [hunk 1]
```diff
@@ -0,0 +1,116 @@
+Handlers patterns
+=================
+
+When you create a new Bottery project, it will contain a file called `handlers.py`.
+This file contains the rules that allow the bot to decide the response to a message,
```
Comment (rougeth) [inline docs/handlers.rst]: "...bot to decide which view will respond the messages"

Comment (rennerocha) [reply docs/handlers.rst]: :+1:

**Comment by rennerocha** [comment]
Is the _patterns.rst_ page still valid? Or it should be dropped from the docs?

File: docs/handlers.rst [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    from bottery import handlers
+
+    from views import pong
+
+    msghandlers = [
+        handlers.message('ping', pong)
+    ]
+
+If the bot receives a message is exactly equal to `ping` the response will be the return value
+of the `pong` function.
```
Comment (rougeth) [inline docs/handlers.rst]: "...exactly equal to..."

File: docs/handlers.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+Bottery has the following handlers available:
+
+handlers.message
+----------------
+
+The message handler allows your bot to check if a message was received. This way, **bottery**
+will check if that exact message was sent by the user. If the message was send exactly like
+you defined it, then the function is called to return a message.
```
Comment (rougeth) [inline docs/handlers.rst]: `you defined it, then the view is used.`

File: docs/handlers.rst [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+The default case is the one that comes when you first created your project:
```
Comment (rougeth) [inline docs/handlers.rst]: "This is the handler that comes on `handlers.py` when you create a project""

File: docs/handlers.rst [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+.. code-block:: py
+
+    def numbers(message):
+        return 'You sent only numbers!'
+
+    msghandlers = [
+        handlers.regex('\d+', numbers),
+    ]
+
+Processing order
```
Comment (rougeth) [inline docs/handlers.rst]: Is "Handling order" better?

Comment (rennerocha) [reply docs/handlers.rst]: Much better

File: docs/handlers.rst [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+    msghandlers = [
+        handlers.startswith('hello', how_are_you)
+    ]
+
+With this pattern, if you receive a message such as `hello, my bot!` it will call our function and
+return an answer.
+
+You should remember that this pattern **is not case-sensitive**! If you wish to
+make a case-sensitive pattern, include the `case_sensitive` parameter:
```
Comment (rougeth) [inline docs/handlers.rst]: This last two sentences need update too

File: docs/handlers.rst [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+    msghandlers = [
+        handlers.startswith('hello', how_are_you, case_sensitive=True)
+    ]
+
+
+handlers.regex
+--------------
+
+The *regex* handler allows you to process a message that matches to a regex pattern.
+If there is a positive match, the view will be executed.
```
Comment (rougeth) [inline docs/handlers.rst]: Also here

**Comment by rougeth** [comment]
@rennerocha almost done! Need to update branch and remove reference of patterns.rst on `docs/index.rst:19`, which is broken on CI.

**Comment by rougeth** [comment]
@rennerocha thank you very much!

**Pull request merged**